tests/indexes/tests.py

```diff
@@ -56,6 +56,36 @@ def test_index_name(self):
             self.skipTest('This test is only supported on the built-in database backends.')
         self.assertEqual(index_name, expected[connection.vendor])
 
+    def test_index_name_multibyte(self):
+        """
+        Assume utf-8 encoding when truncating index names.
+        """
+        # Choose names that only slightly exceed the max name length per backend.
+        # Too long, and multibyte characters wouldn't make a difference.
+        long_name_for_backend = {
+            'mysql': 'I♥Django' * 4,
+            'oracle': 'I♥D',
+            'postgresql': 'I♥Django' * 4,
+            'sqlite': 'I♥Django' * 17,
+        }
```

Comment:
This is not friendly for 3rd-party backends. We should use `connection.ops.max_name_length()` and generate names that are too long.